faucet/valve.py

```diff
@@ -612,6 +616,15 @@ def _update_stack_link_state(self, ports, now, other_valves):
                     ofmsgs_by_valve[valve].extend(valve.flood_manager.add_vlan(vlan))
                 for port in valve.dp.stack_ports:
                     ofmsgs_by_valve[valve].extend(valve.host_manager.del_port(port))
+
+                if valve.flood_manager.graph and not graph_updated:
+                    with open('/etc/faucet/dp_graph.json', 'w') as outfile:
+                        outfile.write(json.dumps(networkx.json_graph.node_link_data(valve.flood_manager.graph), default=str))
+                    self._notify(
+                            {'STACK_TOPO_CHANGE': {
+                                'stack_root': valve.dp.stack_root_name,
+                                'graph': json.dumps(networkx.json_graph.node_link_data(valve.flood_manager.graph), default=str)}})
```

Comment:
Why encode graph as string?  At this point, I think we should be sending through the actual data (which then gets encoded as json by notify)...